truss/truss_config.py

```diff
@@ -324,51 +321,78 @@ def from_dict(d):
     @staticmethod
     def from_yaml(yaml_path: Path):
         with yaml_path.open() as yaml_file:
-            return TrussConfig.from_dict(yaml.safe_load(yaml_file))
+            test = yaml.safe_load(yaml_file)
+            return TrussConfig.from_dict(test)
 
     def write_to_yaml_file(self, path: Path):
         with path.open("w") as config_file:
-            yaml.dump(self.to_dict(), config_file)
+            yaml.dump(self.to_dict(verbose=True), config_file)
 
-    def to_dict(self):
-        d = {
-            "model_type": self.model_type,
-            "model_framework": self.model_framework.value,
-            "model_module_dir": self.model_module_dir,
-            "model_class_filename": self.model_class_filename,
-            "model_class_name": self.model_class_name,
-            "data_dir": self.data_dir,
-            "input_type": self.input_type,
-            "model_metadata": self.model_metadata,
-            "requirements": self.requirements,
-            "system_packages": self.system_packages,
-            "environment_variables": self.environment_variables,
-            "resources": self.resources.to_dict(),
-            "python_version": self.python_version,
-            "model_name": self.model_name,
-            "examples_filename": self.examples_filename,
-            "secrets": self.secrets,
-            "description": self.description,
-            "bundled_packages_dir": self.bundled_packages_dir,
-            "external_package_dirs": self.external_package_dirs,
-            "live_reload": self.live_reload,
-            "spec_version": self.spec_version,
-            "apply_library_patches": self.apply_library_patches,
-            "train": self.train.to_dict(),
-        }
-        if self.external_data is not None:
-            d["external_data"] = transform_optional(
-                self.external_data, lambda data: data.to_list()
-            )
-        if self.base_image is not None:
-            d["base_image"] = transform_optional(
-                self.base_image, lambda data: data.to_dict()
-            )
-        return d
+    def to_dict(self, verbose: bool = True):
```

Comment:
can you please default `verbose` to False and only overwrite it when writing the build context?

Reply:
`TrussConfig.to_dict()` is used in a bunch of places under the assumption that this function returns the fully specified config dictionary – @varunshenoy if you are to switch the default to `False` you will need to find all the places that operate under this assumption and explicitly pass `verbose=True`. This is fine, but @bolasim is there a reason to default to `verbose=False`?

Reply:
I'm fine with this. no good reason.